src/iconv.cpp

```diff
@@ -110,7 +110,7 @@ IConv::convert(const std::string& text)
 
     // We try to avoid to much copying around, so we write directly into
     // a std::string
-    tinygettext_ICONV_CONST char* inbuf = const_cast<char*>(&text[0]);
+    const char* inbuf = const_cast<char*>(&text[0]);
```

Comment:
There is a whole #ifdef section for this in https://github.com/maths22/tinygettext/blob/master/include/tinygettext/iconv.hpp#L25, you should fix it there. Which iconv are you actually using?